metadata-ingestion/src/datahub/utilities/sql_parser.py

```diff
@@ -72,3 +83,116 @@ def get_columns(self) -> List[str]:
                     ]
         # swap back renamed date column
         return ["date" if c == self._DATE_SWAP_TOKEN else c for c in filtered_cols]
+
+
+class SqlLineageSQLParser(SQLParser):
+    _DATE_SWAP_TOKEN = "__d_a_t_e"
+    _TIMESTAMP_SWAP_TOKEN = "__t_i_m_e_s_t_a_m_p"
+
+    def __init__(self, sql_query: str) -> None:
+        super().__init__(sql_query)
+
+        original_sql_query = sql_query
+
+        # MetadataSQLParser makes mistakes on lateral flatten queries, use the prefix
+        if "lateral flatten" in sql_query:
+            sql_query = sql_query[: sql_query.find("lateral flatten")]
+
+        # MetadataSQLParser also makes mistakes on columns called "date", rename them
+        sql_query = re.sub(
+            r"(\bdate\b)", rf"{self._DATE_SWAP_TOKEN}", sql_query, flags=re.IGNORECASE
+        )
+
+        # SqlLineageParser also makes mistakes on columns called "timestamp", rename them
+        sql_query = re.sub(
+            r"(\btimestamp\b)",
+            rf"{self._TIMESTAMP_SWAP_TOKEN}",
+            sql_query,
+            flags=re.IGNORECASE,
+        )
+
+        # MetadataSQLParser does not handle "encode" directives well. Remove them
+        sql_query = re.sub(r"\sencode [a-zA-Z]*", "", sql_query, flags=re.IGNORECASE)
+
+        # Replace lookml templates with the variable otherwise sqlparse can't parse ${
+        sql_query = re.sub(r"(\${)(.+)(})", r"\2", sql_query)
+        if sql_query != original_sql_query:
+            logger.debug(f"rewrote original query {original_sql_query} as {sql_query}")
+
+        self._sql = sql_query
+
+        self._stmt = [
+            s
+            for s in sqlparse.parse(
+                # first apply sqlparser formatting just to get rid of comments, which cause
+                # inconsistencies in parsing output
+                sqlparse.format(
+                    self._sql.strip(),
+                    strip_comments=True,
+                    use_space_around_operators=True,
+                ),
+            )
+            if s.token_first(skip_cm=True)
+        ]
+
+        with unittest.mock.patch(
```

Comment:
Couldn't this be done without using `unittest` and mocking?

Reply:
Currently Sqllineage can't get tables/columns for select queries. 
This is only needed until we contribute back this and sqllineage can get source columns for select queries.

Reply:
This change prevents some companies from using this library.

Imho, better to leave out the column lineage than to use mocking and testing libraries in production code. Or even wait until the related fix is implemented in sqllineage.